compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AMDAIETargetLdScript.cpp

```diff
@@ -95,9 +67,33 @@ SECTIONS
      *(.data*);
      *(.rodata*)
   } > data
+  .comment : {
+     *(.comment*)
+  }
+  .symtab : {
+     *(.symtab)
+  }
+  .shstrtab : {
+     *(.shstrtab)
+  }
+  .strtab : {
+     *(.strtab)
+  }
+  .tctmemtab : {
+     *(.tctmemtab)
+  }
+  .rtstab : {
+     *(.rtstab)
+  }
+  .eoltab : {
+     *(.eoltab)
+  }
+  .chesstypeannotationtab : {
+     *(.chesstypeannotationtab)
+  }
```

Comment:
Orthogonal to this PR specifically, but are all of these needed? I.e. aren't some of these metadata and wouldn't we be able to make the elf smaller by excluding those?

Reply:
The way this works is with these sections specified explicitly, *if* any of the ELFs contain these sections, *then* those sections will be placed at these locations. E.g., this PR failed at first because I turned on `=error` and `mm.o` had a `.chesstypeannotationtab` section that currently gets placed at an arbitrary location. 

So the sections are already there and are already getting placed into the ELF. `-Wl,--gc-sections` pertains to your question and there I do not know what counts as "unused" and therefore "will be gced".

Reply:
I see, thanks for the explanation! I am a rookie on ld scripts so will take your word for it :) At some point, it might be interesting to see what sections are actually copied into the elf.

Reply:
I ain't no pro either - 90% of what I'm saying is just stuff I'm absorbing by osmosis by reading PRs/discussions in llvm-aie.

Reply:
Actually that's not true - I did actually read some "docs" on it (shocking I know); I read about half of this thing https://mcyoung.xyz/2021/06/01/linker-script/.